test/type/self.swift

```diff
@@ -46,6 +46,8 @@ final class FinalMario : Mario {
 // These references to Self are now possible (SE-0068)
 
 class A<T> {
+  typealias _Self = Self
+  // expected-error@-1 {{'Self' is only available in a protocol or as the result of a method in a class; did you mean 'A'?}}
```

Comment:
Can you also add a test for subscripts taking and returning Self? Neither should be allowed (eventually we could allow get-only subscripts to return Self though, along with get-only properties)

Reply:
Done.